CMakeLists.txt

```diff
@@ -923,6 +923,7 @@ add_subdirectory(build_tools/third_party/libyaml EXCLUDE_FROM_ALL)
 
 add_subdirectory(build_tools/third_party/llvm-project EXCLUDE_FROM_ALL)
 add_subdirectory(build_tools/third_party/tracy_client EXCLUDE_FROM_ALL)
+add_subdirectory(build_tools/third_party/tsl EXCLUDE_FROM_ALL)
```

Comment:
I don't see how this can work: tsl doesn't have a CMake build and I don't see one here.

Also, side note, this piqued my curiosity and I was looking through TSL. I couldn't find an oss repo that was accepting contributions for it, and it appears to transitively depend on most of the old tf deps. Both are non starters.

Reply:
Yeah, I missed build_tools/third_party/xsl/CMakeLists.txt with a target to compile xplane.proto in this PR.

Reply:
TSL is being proposed to be merged into XLA, doesn't address the cmake issue.

Reply:
Or the deps/api. A tracing event API should have a narrow interface and requirements. Think: "could I use this reasonably from a Rust app" or "would Tracy accept this as a connector" and you'll have a similar analysis of what it should look like to integrate here.

Again, if there is utility we want it in some fashion, but we could use some education on that. In its current proposed form, it would add complexity of a similar scale to taking an LLVM dep... And that is a lot for something that needs to assemble a ~5 field proto message and squirt it to a receiver. The last two of those that I've written had literally zero deps so that they could go anywhere, and you see systems like Tracy and cupti following suit for similar reasons by providing a very narrow integration surface.

Reply:
(it may also be possible to invert the situation somehow and make iree expose apis that fit with sinks like this but I'd need to know more about what is trying to be accomplished)